include/api/CDataFrameAnalyzer.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+#ifndef INCLUDED_ml_api_CDataFrameAnalyzer_h
+#define INCLUDED_ml_api_CDataFrameAnalyzer_h
+
+#include <core/CLogger.h>
```

Comment:
You don't need the logger in the header file - this line should be in the `.cc` file instead.